DOCS-2845: update afterpay, e-invoicing, in3, klarna, pay after delivery testing details (#1826)
MultiSafepay/docs: content/payments/integrations/testing/test-payment-details.md

**Test an AfterPay order**

1. Send a [direct or redirect](/api/#afterpay) API request. For more information, see [Difference between direct and redirect API requests](/developer/api/difference-between-direct-and-redirect).

> 2. If you send a redirect API request, select the checkbox at the bottom of the AfterPay page, and then click **Confirm**.  

Reviewer: ```suggestion
2. If you make a redirect API request, select the checkbox at the bottom of the AfterPay page, and then click **Confirm**.  
```

**Note:** You can't test:  

- Receiving successful payment notifications from AfterPay
- Changing the transaction status from **Uncleared** to **Completed**
- Processing refunds

{{< /details >}}

{{< details title="Betaal per Maand" >}}

You can't test Betaal per Maand in your MultiSafepay test account. You can only make test payments in your MultiSafepay live account.

{{< /details >}}

{{< details title="E-Invoicing" >}}

Test credentials: [API key](/account/site-id-api-key-secure-code/)

**Test an E-Invoicing order**

> To test an E-Invoicing order, send a [direct](/api/#e-invoicing---direct) or [redirect](/api/#e-invoicing---redirect) API request.

Reviewer: ```suggestion
To test an E-Invoicing order, make a [direct](/api/#e-invoicing---direct) or [redirect](/api/#e-invoicing---redirect) API request.
```

> If you send a redirect API request:

Reviewer: ```suggestion
If you make a redirect API request:
```

- Enter in the:
  - **Birthdate** field any date of birth. Format: DD-MM-YYYY.
  - **Bank account** field any 10-digit bank account number.
  - **Email address** field any email address.
  - **Phone number** field any phone number.
- Click **Confirm**.

The payment is processed in the test environment as **Successful**, with order and transaction statuses **Uncleared**.

**Test declining an order**  

To decline an order, in your test account under **Order summary**, click **Decline**.  
The order and transaction statuses change to **Void**.

**Test shipping an order**  

> To test shipping an order, send an [update an order](/api/#update-an-order) API request with status `"shipped"`. You receive the `invoice_url` in the API response.

Reviewer: ```suggestion
To test shipping an order, make an [update an order](/api/#update-an-order) API request with status `"shipped"`. You receive the `invoice_url` in the API response.
```

---

**Note:** 
You can't test:
- Processing refunds
- Cancelling orders

{{< /details >}}

{{< details title="in3" >}}

Test credentials: [API key](/account/site-id-api-key-secure-code/)

**Test an in3 order**

> 1. To test an in3 order, send a [direct](/api/#in3---direct) or [redirect](/api/#in3---redirect) API request with the following customer details.

Reviewer: ```suggestion
1. To test an in3 order, make a [direct](/api/#in3---direct) or [redirect](/api/#in3---redirect) API request with the following customer details.
```

| Date of birth    | Postal code | House number |
| ------------------- | ------------------- | ----------------- |
| 01-01-1999 | 1234AB | 1 |

>     If you send a redirect API request:

Reviewer: ```suggestion
    If you make a redirect API request:
```

- Enter in the:
    - **Birthdate** field `01-01-1999`.
    - **Phone number** field any phone number.  
- Select your title, and then click **Confirm**.
2. Select the checkbox to accept in3's payment terms and privacy statement, and then click **Afronden**.
3. On the Test platform page, from the **Test scenario** list, select **Completed**.
4. Click **Test**. 
5. On the in3 page, click **Terug naar webshop**.  
  The payment is processed in the test environment as **Successful**, with order status **Completed**, and transaction status **Uncleared**.

**Test in3 declining an order**  

> To test in3 declining an order, send a [direct](/api/#in3---direct) or [redirect](/api/#in3---redirect) API request with the following customer details:

Reviewer: ```suggestion
To test in3 declining an order, make a [direct](/api/#in3---direct) or [redirect](/api/#in3---redirect) API request with the following customer details:
```

| Date of birth    | Postal code | House number |
| ------------------- | ------------------- | ----------------- |
| 01-01-2000 | 1111AB | 1 |

The order and transaction statuses change to **Declined**.

**Test shipping an order**  
To test shipping an order, either:

> - Send an [update an order](/api/#update-an-order) API request with status `shipped`, or 

Reviewer: ```suggestion
- Make an [update an order](/api/#update-an-order) API request with status `shipped`, or 
```

- In your MultiSafepay test account, go to **Order summary**, and then click **Order status**.

**Receive an invoice**  

You can only test invoicing in your MultiSafepay live account. To do this, change the order status to **Shipped**.

**Test refunding an order**

To test refunding an order:

1. Create an order. 
2. Change the order status to `shipped`.
3. Click **Refund complete order**, and then click **Save item changes**.
  {{< br >}} A new order is created for the refund. The order status for the refund changes to **Completed**.

**Test an API refund**

To test refunding an order via the API:

1. Create an order. 
2. Change the order status to `shipped`.

> 3. Send a [refund with shopping cart](/api/#refund-with-shopping-cart) API request.

Reviewer: ```suggestion
3. Make a [refund with shopping cart](/api/#refund-with-shopping-cart) API request.
```

**Note:** You can't test:  

- Receiving successful payment notifications from AfterPay
- Changing transaction statuses from **Uncleared** to **Completed**
- Processing refunds

{{< /details >}}

{{< details title="Betaal per Maand" >}}

You can't test Betaal per Maand in your MultiSafepay test account. You can only make test payments in your MultiSafepay live account.

{{< /details >}}

{{< details title="E-Invoicing" >}}

Test credentials: [API key](/account/site-id-api-key-secure-code/)

**Test an E-Invoicing order**

To test an E-Invoicing order, send a [direct](/api/#e-invoicing---direct) or [redirect](/api/#e-invoicing---redirect) API request.

> If you send a redirect API request, enter in the:

Reviewer 2: ```suggestion
If you send a redirect API request:  
- Enter in the:
```

  - **Birthdate** field any date of birth. Format: DD-MM-YYYY.
  - **Bank account** field any 10-digit bank account number.
  - **Email address** field any email address.
  - **Phone number** field any phone number.

>   Click **Confirm**.

Reviewer 2: ```suggestion
- Click **Confirm**.
```

The payment is processed in the test environment as **Successful**, with order and transaction statuses **Uncleared**.

**Test declining an order**  

To decline an order, in your test account under **Order summary**, click **Decline**.  
The order and transaction statuses change to **Void**.

**Test cancelling an order**

To test cancelling an order:

1. Send a [direct](/api/#e-invoicing---direct) API request.
2. Either:
    - Send an [update an order](/api/#update-an-order) API request with status `"cancelled"`, or 
    - In your MultiSafepay test account, go to **Order summary**, click **Order status**.

>     - From **Change status to**, select **cancelled**, in the **memo** field enter a reason, and then click **Ok**.  

Reviewer 2: ```suggestion
    - From **Change status to**, select **Cancelled**.  
    - In the **Memo** field, enter a reason.  
    - Click **Ok**.  
```

  The transaction status changes to **Void** and the order status changes to **Cancelled**.

**Test shipping an order**  

To test shipping an order, send an [update an order](/api/#update-an-order) API request with status `"shipped"`. You receive the `invoice_url` in the API response.

---

**Note:** You can't test processing refunds.

{{< /details >}}

{{< details title="in3" >}}

Test credentials: [API key](/account/site-id-api-key-secure-code/)

**Test an in3 order**

1. To test an in3 order, send a [direct](/api/#in3---direct) or [redirect](/api/#in3---redirect) API request with the following customer details.
| Date of birth    | Postal code | House number |
| ------------------- | ------------------- | ----------------- |
| 01-01-1999 | 1234AB | 1 |

> - If you send a redirect API request, enter in the:

Reviewer 2: ```suggestion
If you send a redirect API request:  
- Enter in the:  
```

    - **Birthdate** field `01-01-1999`.
    - **Phone number** field any phone number.  

>   Select your title, and then click **Confirm**.

Reviewer 2: ```suggestion
- Select your title, and then click **Confirm**.
```

2. Select the checkbox confirming that you accept in3's payment terms and privacy statement, and then click **Afronden**.
3. On the Test platform page, from the **Test scenario** list, select **Completed**.
4. Click **Test**. 
5. On the in3 page, click **Terug naar webshop**.  
  The payment is processed in the test environment as **Successful**, with order status **Completed**, and transaction status **Uncleared**.

**Test in3 declining an order**  

To test in3 declining an order, send a [direct](/api/#in3---direct) or [redirect](/api/#in3---redirect) API request with the following customer details:

| Date of birth    | Postal code | House number |
| ------------------- | ------------------- | ----------------- |
| 01-01-2000 | 1111AB | 1 |

The order and transaction statuses change to **Declined**.

**Test shipping an order**  
To test shipping an order, either:

- Send an [update an order](/api/#update-an-order) API request with status `shipped`, or 
- In your MultiSafepay test account, go to **Order summary**, and then click **Order status**.

**Receive an invoice**  

> You can only test the invoice process in your MultiSafepay live account. To do this, change the order status to **Shipped**.

Reviewer 2: ```suggestion
You can only test invoicing in your MultiSafepay live account. To do this, change the order status to **Shipped**.
```

**Test refunding an order**

To test refunding an order:

1. Create an order. 
2. Change the order status to `shipped`.
3. Click **Refund complete order**, and then click **Save item changes**.
  {{< br >}} A new order is created for the refund. The order status for the refund changes to **Completed**.

**Test an API refund**

To test refunding an order via the API:

1. Create an order. 
2. Change the order status to `shipped`.
3. Send a [refund with shopping cart](/api/#refund-with-shopping-cart) API request.
  {{< br >}} A new order is created for the refund. The order status for the refund changes to **Completed**.

---

> **Note:** 
> You can't test cancelling orders.

Reviewer 2: ```suggestion
**Note:** You can't test cancelling orders.
```

- [Klarna's test credentials](https://docs.klarna.com/resources/test-environment/)

**Test a Klarna order**  

> 1. Send a [direct](/api/#klarna) API request. 

Reviewer: ```suggestion
1. Make a [direct](/api/#klarna) API request. 
```

**Receive an invoice**  

You can only test invoicing in your MultiSafepay live account. To do this, change the order status to **Shipped**.

---

**Note:** You can't test:

- Receiving successful payment notifications from Klarna
- Changing the transaction status from **Uncleared** to **Completed**, except for refunds
- Sending redirect API requests

For more information about integrating Klarna with MultiSafepay, see Payment methods – [Klarna](/payments/methods/billing-suite/klarna).
{{< /details >}}

{{< details title="Pay After Delivery (Betaal na Ontvangst)" >}}

Test credentials: [API key](/account/site-id-api-key-secure-code/)

**Test a Pay After Delivery order**

> To test a Pay After Delivery order, send a [direct](api/#pay-after-delivery---direct) or [redirect](api/#pay-after-delivery---redirect) API request.

Reviewer: ```suggestion
To test a Pay After Delivery order, make a [direct](api/#pay-after-delivery---direct) or [redirect](api/#pay-after-delivery---redirect) API request.
```

> If you send a redirect API request, click **Pay After Delivery**.  

Reviewer: ```suggestion
If you make a redirect API request, click **Pay After Delivery**.  
```

**Receive an invoice**  

> You can only test the invoice process in your MultiSafepay live account. To do this, change the order status to **Shipped**.

Reviewer 2: ```suggestion
You can only test invoicing in your MultiSafepay live account. To do this, change the order status to **Shipped**.
```

---

**Note:** You can't test:

- Receiving successful payment notifications from Klarna
- Changing transaction statuses from **Uncleared** to **Completed**, except for refunds
- Sending redirect API requests

For more information about integrating Klarna with MultiSafepay, see Payment methods – [Klarna](/payments/methods/billing-suite/klarna).
{{< /details >}}

{{< details title="Pay After Delivery (Betaal na Ontvangst)" >}}

Test credentials: [API key](/account/site-id-api-key-secure-code/)

**Test a Pay After Delivery order**

To test a Pay After Delivery order, send a [direct](api/#pay-after-delivery---direct) or [redirect](api/#pay-after-delivery---redirect) API request.

> If you send a redirect API request, click **Pay After Delivery**.  Enter in the:

Reviewer 2: ```suggestion
If you send a redirect API request, click **Pay After Delivery**.   
- Enter in the:
```

  - **Birthdate** field any date of birth. Format: DD-MM-YYYY.
  - **Bank account** field any 10-digit bank account number.
  - **E-mail address** field any email address.
  - **Phone number** field any phone number.  

> Click **Confirm**.

Reviewer 2: ```suggestion
- Click **Confirm**.
```

The payment is processed in the test environment as **Successful**, with order status **Completed**, and transaction status **Uncleared**.

**Test declining an order**  

To decline an order, in your test account under **Order summary**, click **Decline**.  
The order and transaction statuses change to **Void**.

**Test cancelling an order**

To test cancelling an order:

1. Create an order.
2. In your MultiSafepay test account, go to **Order summary**, click **Order status**.

> 3. From **Change status to**, select **cancelled**, in the **memo** field enter a reason, and then click **Ok**.  

Reviewer 2: ```suggestion
3. From **Change status to**, select **Cancelled**.  
4. In the **Memo** field, enter a reason.
5. Click **Ok**.  
```

  The order status changes to **Cancelled**, and the transaction status changes to **Void**.

---

**Note:** 
You can't test:  
  - Processing refunds

>   - Changing the statuses of orders to shipped

Reviewer 2: ```suggestion
  - Changing the order status to **Shipped**
```
